### ISSUE TITLE
Run baseline benchmark if it doesn't exist

### ISSUE THREAD

**Issue by jonkeane** [conbench/conbench#67]
Sometimes the baseline benchmarks haven't been run (either they weren't in the benchmark suite when that commit was pushed, or they weren't run on a specific machine for some reason, etc.). Would it be possible to run any baseline benchmarks that aren't present when a benchmark command is issued from a PR?

This would also enable us to be able to ask for a broader array of benchmark options on demand (e.g. extra compression options) and have a comparison for them, while _generally_ only running the base/default cases.

**Comment by jgehrcke** [comment]
For regression detection not only the 'contender' may potentially need to be re-run, but also the baseline run. If the baseline doesn't exist in the first place then that might be a special case of "re"running it.

**Comment by austin3dickey** [comment]
I like this idea but it feels a bit beyond the scope of what the Conbench runner tools can/should do. Different repos have different ways of setting up benchmark suites on a given commit. Running the baseline again could be as easy as doing a "git checkout <baseline_sha>" and running benchmarks again, but it probably is a lot more custom.

If we go ahead with this, the API would probably be something like: `benchconnect` tells the CI machine it needs to re-run baselines, and the CI machine sets that up, and then `benchconnect` continues onward. Will need some careful thought.

**Comment by jonkeane** [comment]
Agreed that this is actually probably the wrong location for this issue (and would be quick complicated to solve at the conbench server level). Should we move this over to https://github.com/voltrondata-labs/arrow-benchmarks-ci instead? I think it would be _much easier_ to orchestrate from there if we were to decide to do this.

**Comment by austin3dickey** [comment]
I think that would be a great start. If we figure out a good way to do it there, we could either write a playbook for other projects to implement something like this, or somehow abstract it further into code for this repo.
